cdk/src/app-github-actions.ts

```diff
@@ -0,0 +1,99 @@
+import 'source-map-support/register.js'
+import * as cdk from 'aws-cdk-lib'
+import * as iam from 'aws-cdk-lib/aws-iam'
+import { Construct } from 'constructs'
+
+const app = new cdk.App({
+  context: {
+    name: 'github-actions',
+  },
+})
+
+type RepoIsh = `${string}/${string}`
+
+type GitHubActionsStackProps = cdk.StackProps & {
+  repo: RepoIsh
+}
+
+/**
+ * Stack created to bootstrap deployment AWS Account with the necessary resources for deploying from GitHub Actions
+ *
+ * Resources:
+ * - https://dev.to/simonireilly/secure-aws-cdk-deployments-with-github-actions-3jfk
+ * - https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+ */
+export class GitHubActionsStack extends cdk.Stack {
+  private readonly appName: string = this.node.tryGetContext('name')
+
+  constructor(scope: Construct, id: string, props: GitHubActionsStackProps) {
+    super(scope, id, props)
+
+    const { repo } = props
+
+    cdk.Tags.of(this).add('app:name', this.appName)
+
+    /**
+     * Create an Identity provider for GitHub inside your AWS Account. This
+     * allows GitHub to present itself to AWS IAM and assume a role.
+     */
+    const provider = new iam.OpenIdConnectProvider(
+      this,
+      'GitHubActionsOidcProvider',
+      {
+        url: 'https://token.actions.githubusercontent.com',
+        clientIds: ['sts.amazonaws.com'],
+      }
+    )
+
+    /**
+     * Create a principal for the OpenID; which can allow it to assume
+     * deployment roles.
+     */
+    const GitHubPrincipal = new iam.OpenIdConnectPrincipal(
```

Comment:
Super nit. In my repo, I would enforce this, but I don't believe TS has as strong of conventions.

```suggestion
    const gitHubPrincipal = new iam.OpenIdConnectPrincipal(
```

Reply:
good callout, to keep the convention consistent I'll update this to just `principal`